### PR TITLE
Fix/UMA 4635 check image size image

### PR DESCRIPTION
**Pull request by juananime**
## Checklist

- [ ] Create feature in FastImage that throws an error if remote image is not loaded or image size is 1 pixel by 1 pixel.  

## Test
Add FastImage plugin to RN project.
``import FastImage from "@bam.tech/react-native-fast-image";``

Add FastImage to render process:
``< FastImage onError={this.onLoadError} source={"https://img.uefa.com/imgml/TP/teams/logos/700x700/5003e7.png"} resizeMode={"contain"} />;``

Assing onError to method
``onLoadError = () => {
   console.log("image is either not loading or too small");
  };``

Once compiled, the console must show the error message.

### CODE REVIEW

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java [hunk 1]
```diff
@@ -174,4 +232,4 @@ public float getGranularityPercentage() {
         return 0.5f;
     }
 
-}
+}
```
Comment (tychota) [inline android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java]: Missing EOL

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java [hunk 2]
```diff
@@ -41,19 +40,15 @@ public void run() {
...
-                            .diskCacheStrategy(DiskCacheStrategy.ALL)
-                            .priority(priority);
-
                     Glide
                             .with(activity.getApplicationContext())
-                            .applyDefaultRequestOptions(options)
                             .load(glideUrl)
+                            .priority(priority)
+                            .placeholder(TRANSPARENT_DRAWABLE)
+                            .diskCacheStrategy(DiskCacheStrategy.SOURCE)
```
Comment (tychota) [inline android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java:48]: `DisCacheStrategy.ALL`

File: ios/FastImage/FFFastImageView.m [hunk 3]
```diff
@@ -12,6 +14,27 @@ - (void)setResizeMode:(RCTResizeMode)resizeMode
     }
 }
 
+- (void)setOnFastImageLoadEnd:(RCTBubblingEventBlock)onFastImageLoadEnd {
```
Comment (tychota) [inline ios/FastImage/FFFastImageView.m:17]: I would have like a different commit for ios stuff. It is easier to read as a reviewer.

File: ios/FastImage/FFFastImageViewManager.m [hunk 4]
```diff
@@ -14,6 +14,11 @@ - (FFFastImageView*)view {
   return view;
 }
 
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
```
Comment (tychota) [inline ios/FastImage/FFFastImageViewManager.m:19]: Top !!

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java [hunk 5]
```diff
@@ -44,16 +43,13 @@ public ImageViewWithUrl(Context context) {
...
-
+    private static final String REACT_ON_ERROR_EVENT = "onFastImageError";
+    private static final String REACT_ON_LOAD_EVENT = "onFastImageLoad";
+    private static final String REACT_ON_LOAD_END_EVENT = "onFastImageLoadEnd";
     private static final Drawable TRANSPARENT_DRAWABLE = new ColorDrawable(Color.TRANSPARENT);
-
-    private static final Map<String, List<ImageViewWithUrl>> VIEWS_FOR_URLS = new WeakHashMap<>();
-
-    private static RequestManager requestManager = null;
+    private static final Map<String, List<ImageViewWithUrl>> VIEWS_FOR_URLS = new HashMap<>();
```
Comment (tychota) [inline android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java]: An't you leaking the whole view ? seem's like a big memory leak !!

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java [hunk 6]
```diff
@@ -62,15 +58,56 @@ public String getName() {
...
+                Target<GlideDrawable> target,
+                boolean isFirstResource
+        ) {
+            OkHttpProgressGlideModule.forget(uri.toStringUrl());
+            if (!(target instanceof ImageViewTarget)) {
+                return false;
+            }
+            ImageViewWithUrl view = (ImageViewWithUrl) ((ImageViewTarget) target).getView();
+            ThemedReactContext context = (ThemedReactContext) view.getContext();
+            RCTEventEmitter eventEmitter = context.getJSModule(RCTEventEmitter.class);
```
Comment (tychota) [inline android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java:80]: Why use `RCTEventEmitter` and not `RCTNativeAppEventEmitter` ? 
Maybe it is right but I'm not sure why. Most modules tends to use either `RCTDeviceEventEmitter` or recently `RCTNativeAppEventEmitter`

File: android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java [hunk 7]
```diff
@@ -64,15 +65,12 @@ public static void expect(String key, ProgressListener listener) {
...
-
         void update(String key, long bytesRead, long contentLength);
     }
 
     private static class DispatchingProgressListener implements ResponseProgressListener {
-
-        private static final Map<String, ProgressListener> LISTENERS = new WeakHashMap<>();
-
-        private static final Map<String, Long> PROGRESSES = new WeakHashMap<>();
+        private static final Map<String, ProgressListener> LISTENERS = new HashMap<>();
```
Comment (tychota) [inline android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java]: Did'nt we leak the ProgressListener class here ?

Comment (tychota) [reply android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java]: Should not be static, isn't it ?

Comment (tychota) [reply android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java]:
```java
private final Map<String, ProgressListener> LISTENERS = new WeakHashMap<>();
private final Map<String, Long> PROGRESSES = new HashMap<>();
```

**Review by tychota — CHANGES_REQUESTED** [review]
Please be sure to apply the `ARGB_8888`codec:

https://gist.github.com/kevalpatel2106/de101a4dd7d2f3f1a4cb1e22d1d078ba#file-androidmenifest-xml

Read more about this here https://medium.com/@multidots/glide-vs-picasso-930eed42b81d

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java (hunk 5 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java]: Yes you are right, rswitching HashMap for HashMapWeak  Next commit will be amended

File: android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java (hunk 7 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java]: Same, issue as the other comment, removing static for next push.

File: ios/FastImage/FFFastImageView.m (hunk 3 above)
Comment (juananime) [reply ios/FastImage/FFFastImageView.m:17]: Ok, will do next PR. Sorry for that.

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java (hunk 2 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java:48]: will add next commit

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java (hunk 5 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java]: Yes you are right. Switching to private next commit .

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java (hunk 1 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java]: Adding next commit

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java (hunk 2 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java]: Will add next commit

File: android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java (hunk 7 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java]: I reckon there is another method below that removes the instances form the Map, therefor is it actually removed from memory and not leaked?

 static void forget(String key) {
            LISTENERS.remove(key);
            PROGRESSES.remove(key);
        }

File: android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java (hunk 5 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/FastImageViewManager.java]: Added GlideConfiguration for next push

File: android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java (hunk 7 above)
Comment (juananime) [reply android/src/main/java/com/dylanvann/fastimage/OkHttpProgressGlideModule.java]: Switching Hashmap for HashMapWeak

**Review by tychota — APPROVED** [review]

**Pull request merged**